runtime/context.go

```diff
@@ -57,13 +57,37 @@ except that the forwarded destination is not another HTTP service but rather
 a gRPC service.
 */
 func AnnotateContext(ctx context.Context, mux *ServeMux, req *http.Request) (context.Context, error) {
+	ctx, md, err := annotateContext(ctx, mux, req)
+	if err != nil {
+		return nil, err
+	} else if md == nil {
```

Comment:
Using `else` unnecessarily is often a bit frowned upon. How about 
```suggestion
	}
	if md == nil {
```